src/components/interactive-builder/modals/question/question-form/question/question.test.tsx

```diff
@@ -0,0 +1,197 @@
+    expect(idInput).toHaveValue('testId');
+  });
+
+  it('should update form field when question id changes', () => {
+    renderWithFormFieldProvider(<Question checkIfQuestionIdExists={jest.fn()} />);
+    const idInput = screen.getByRole('textbox', { name: /Question ID/i });
+
+    fireEvent.change(idInput, { target: { value: 'newTestId' } });
+
+    expect(idInput as HTMLInputElement).toHaveValue('newTestId');
+  });
+
+  // it('should validate duplicate question ids', () => {
+  //   const checkIfQuestionIdExists = jest.fn(() => true);
+  //   renderWithFormFieldProvider(<Question checkIfQuestionIdExists={checkIfQuestionIdExists} />, {
+  //     formField: { ...initialFormField, id: 'duplicateId' },
+  //   });
+
+  //   expect(checkIfQuestionIdExists).toHaveBeenCalledWith('duplicateId');
+  //   expect(screen.getByText(/This question ID already exists/i)).toBeInTheDocument();
+  // });
+
+  it('should convert label to camel case when button is clicked', () => {
+    renderWithFormFieldProvider(<Question checkIfQuestionIdExists={jest.fn()} />, {
+      formField: { ...initialFormField, label: 'Test Label For Camel Case' },
+    });
+
+    const convertButton = screen.getByText(/Convert label to camel-case/i);
+    fireEvent.click(convertButton);
+
+    const idInput = screen.getByRole('textbox', { name: /Question ID/i }) as HTMLInputElement;
+    expect(idInput).toHaveValue('testLabelForCamelCase');
+  });
+
+  it('should toggle question info visibility', () => {
+    renderWithFormFieldProvider(<Question checkIfQuestionIdExists={jest.fn()} />);
+
+    expect(screen.queryByLabelText(/Additional Question Info/i)).not.toBeInTheDocument();
+
+    const yesRadio = screen.getByLabelText(/Yes/i);
+    fireEvent.click(yesRadio);
+
+    expect(screen.getByLabelText(/Additional Question Info/i)).toBeInTheDocument();
+
+    const noRadio = screen.getByLabelText(/No/i);
+    fireEvent.click(noRadio);
+
+    expect(screen.queryByLabelText(/Additional Question Info/i)).not.toBeInTheDocument();
+  });
+
+  it('should update question info when input changes', () => {
+    renderWithFormFieldProvider(<Question checkIfQuestionIdExists={jest.fn()} />);
+
+    const yesRadio = screen.getByLabelText(/Yes/i);
+    fireEvent.click(yesRadio);
+
+    const questionInfoInput = screen.getByLabelText(/Additional Question Info/i);
+    fireEvent.change(questionInfoInput, { target: { value: 'New question info' } });
+
+    expect((questionInfoInput as HTMLInputElement).value).toBe('New question info');
+  });
+
+  it('should toggle question required status', () => {
+    renderWithFormFieldProvider(<Question checkIfQuestionIdExists={jest.fn()} />);
+
+    const requiredRadio = screen.getByLabelText(/Required/i);
+    fireEvent.click(requiredRadio);
+
+    // TODO: check if it gets the updated prop
+
+    const optionalRadio = screen.getByLabelText(/Optional/i);
+    fireEvent.click(optionalRadio);
```

Comment:
Prefer to use userEvents over fireEvent - https://testing-library.com/docs/user-event/intro#differences-from-fireevent